hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/scheduler/HdfsScheduler.java

```diff
@@ -127,6 +126,10 @@ public void statusUpdate(SchedulerDriver driver, TaskStatus status) {
       status.getMessage(),
       liveState.getStagingTasksSize()));
 
```

Comment:
1) Moving Observer notification there wouldn't work.  The Reconciler must receive updates of TerminalState Tasks so that it can remove them from its list of pendingTasks (for reconciliation).  If it is reconciling a Task and we get a notification that it was LOST we still need that status update.  That's a reconciled task.

2) The idea going forward is that a large portion of the logic in the statusUpdate method should be moved out of the HdfsScheduler class anyway.  Components which need to take action based on the statusUpdate stream are Observers.  They should behave as the Reconciler does here.  When reconciliation is complete, the Reconciler essentially does a no-op when receiving a statusUpdate.

3) Finally statusUpdates are actually pretty infrequent and small in number.  The overhead of sending updates to "all observers" is very low.